src/toil/test/src/fileStoreTest.py

```diff
@@ -250,6 +252,40 @@ def _testFileStoreOperations(job, nonLocalDir, numIters=100):
                             logger.info('No longer have file: %s', fsID)
                 i += 1
 
+        def testWriteReadGlobalFile(self):
+            workdir = self._createTempDir(purpose='nonLocalDir')
+            for i in True,False:
+                A = Job.wrapJobFn(self._testWriteReadGlobalFilePermissions,
+                                isLocalFile=False, executable=i, nonLocalDir=workdir)
+                Job.Runner.startToil(A, self.options)
+
+        @staticmethod
+        def _testWriteReadGlobalFilePermissions(job, isLocalFile, executable, nonLocalDir=None):
```

Comment:
You should drop `isLocalFile` and `nonLocalDir` from this function, because I don't think we care about testing the exact source directory from which we upload the file for this test.

Just do `workDir = job.fileStore.getLocalTempDir()` always, and always make the file in `workDir`.